src/libraries/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/SyndicationContent.cs

```diff
@@ -85,7 +85,7 @@ public void WriteTo(XmlWriter writer, string outerElementName, string outerEleme
             {
                 foreach (XmlQualifiedName key in _attributeExtensions.Keys)
                 {
-                    if (key.Name == Atom10Constants.TypeTag && key.Namespace == string.Empty)
+                    if (key.Name == Atom10Constants.TypeTag && key.Namespace.Length == 0)
```

Comment:
Ditto

Reply:
The `XmlQualifiedName` constructors seem to handle if `null` was passed. There is a check there that sets it to string.Empty, Is that a kind of guarantee to depend on?

Reply:
But it also looks like those fields can be assigned by the result of calling XmlNameTable.Add, which is an abstract method on a public type.
https://github.com/dotnet/runtime/blob/ec08b855725a1566f2cd66200aaba37ef405c62d/src/libraries/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs#L161-L166
Without the due diligence of going through and validating that there's no possible way this could be null, this is not a safe change.

Reply:
@stephentoub, Sounds good. I'm reverting it back.
Thanks for review, and for patience answering my doubts :)
And excuse me if my questions was many. I'm just trying to learn while contributing :)

```suggestion
                    if (key.Name == Atom10Constants.TypeTag && key.Namespace == string.Empty)
```